scim-sdk-server/src/main/java/de/captaingoldfish/scim/sdk/server/patch/workarounds/msazure/MsAzurePatchComplexValueRebuilder.java

```diff
@@ -157,4 +149,29 @@ public PatchRequestOperation fixPatchRequestOperaton(ResourceType resourceType,
     operation.setValues(fixedValues);
     return operation;
   }
+
+  private boolean shouldBeFixed(String jsonDocument)
+  {
+    if (StringUtils.isBlank(jsonDocument))
+    {
+      return true;
+    }
+
+    Pattern pattern = Pattern.compile(JSON_RESERVED_CHARS_PATTERN);
+    Matcher matcher = pattern.matcher(jsonDocument);
+
+    if (!matcher.find())
+    {
+      jsonDocument = String.format("\"%s\"", jsonDocument);
+    }
+
+    try
+    {
+      JsonNode jsonNode = JsonHelper.readJsonDocument(jsonDocument);
+      return jsonNode != null && !jsonNode.isObject() && !jsonNode.isArray();
+    } catch (IOException ex)
+    {
+      return false;
+    }
```

Comment:
this method should not be needed. It seems you have added it due to the failing unit tests. But the unit tests must be fixed instead. The unit tests should verify that the `shouldBeHandled` method returns false in these specific cases.

Reply:
We can move/restructure the `shouldBeHandled` method as you want, but the pattern part must stay. I extended the unit tests to cover also cases that fails with the current implementation. You can try it. 

The problem here is how Jackson handles JSON strings. If the input is a string like this `String input = "abcd1324"`, then Jackson parses `abcd1324`. It is missing the double quotation. And unfortunately -> `"abcd1324"` is a valid JSON, `abcd1324` is not.

So, I am trying to identify when the value is just a simple string and add the quotation marks. That makes it a valid JSON and your `JsonHelper. readJsonDocument` is happy. I admit the code is a bit hacky, but it works...

Reply:
@Captain-P-Goldfish Can you check my comment, please? ☝️  I am sure it is not just about tests

Reply:
It would help me if you could start a pull-request. It is easier to do a checkout on the pull-request and check what you said. Until now you haven't created the pull-request.

Reply:
I don't have proper permissions to push a branch into your main repo. So I cannot open a PR there. That's why I opened this PR. Not sure how to handle it differently. What would you have me do?

Reply:
if you have forked the repository github should offer you to create a pull-request. Direct commits are not allowed of course. Here is an example that I just made:
![test](https://github.com/user-attachments/assets/3c14348d-3d92-4260-9d9d-5e64582a1ed5)